uniswap_universal_router_decoder/_encoder.py

```diff
@@ -872,6 +883,42 @@ def permit2_transfer_from(
         self._add_command(RouterFunction.PERMIT2_TRANSFER_FROM, args)
         return self
 
+    def permit2_transfer_from_batch(
+            self,
+            transfer_details: List[Permit2TransferDetail]) -> _ChainedFunctionBuilder:
+        """
+        Encode the batch transfer of multiple tokens from the caller address to recipients.
+        The UR must have been permit2'ed for all tokens first.
+
+        :param transfer_details: List of dictionaries with keys: 'from', 'to', 'amount', 'token'
+            Each dict should contain:
+            - 'from': The address to transfer tokens from (usually the caller)
+            - 'to': The recipient address
+            - 'amount': The amount to transfer (in Wei)
+            - 'token': The token address
+        :return: The chain link corresponding to this function call.
+        """
+        # Validate input structure
+        if not transfer_details:
+            raise ValueError("transfer_details cannot be empty")
+        for i, detail in enumerate(transfer_details):
+            required_keys = {"from", "to", "amount", "token"}
+            if not all(key in detail for key in required_keys):
+                raise ValueError(f"Transfer detail {i} missing required keys: {required_keys}")
+
+            # Validate that amounts are positive (Wei is an int-like type)
+            if int(detail["amount"]) <= 0:
+                raise ValueError(f"Transfer detail {i} has non-positive amount: {detail['amount']}")
+
```

Comment:
Please remove the whole validation section (line 901 to 911) as this is rather redundant with a type checker.

A point could be made about the amount sign which is not checked by `mypy`. But a negative amount will be rejected by the underlying ABI encoder library, with a specific error consistent with other errors raised by web3.py. I prefer not interfering with well known exceptions.